src/main/java/org/kuali/kra/subaward/service/impl/SubAwardServiceImpl.java

```diff
@@ -252,15 +252,20 @@ public SubAward getAmountInfo(SubAward subAward) {
         subAward.setTotalAvailableAmount(totalObligatedAmount.subtract(totalAmountReleased));
 
         List<SubAwardAmountInfo> sortedlist = subAward.getSubAwardAmountInfoList();
+
         if ( sortedlist.size() > 1 ) {
             Collections.sort(sortedlist, new Comparator<SubAwardAmountInfo>() {
                 public int compare(SubAwardAmountInfo o1, SubAwardAmountInfo o2) {
-                    return o1.getSubAwardAmountInfoId() - o2.getSubAwardAmountInfoId();
+                    if((o1.getSubAwardAmountInfoId() != null) && (o2.getSubAwardAmountInfoId() != null)) {

```

Comment:
Alex, when you debugged to this point, are you sure that both o1 and o2 are not null at this point? Or should we test that as well?